client/keys/show.go

```diff
@@ -18,13 +22,38 @@ var showKeysCmd = &cobra.Command{
 	RunE: func(cmd *cobra.Command, args []string) error {
 		name := args[0]
 		info, err := getKey(name)
-		if err == nil {
-			printInfo(info)
+		if err != nil {
+			return err
 		}
-		return err
+
+		showAddress := viper.GetBool(client.FlagAddress)
+		showPublicKey := viper.GetBool(client.FlagPublicKey)
+		outputSet := cmd.Flag(cli.OutputFlag).Changed
+		if showAddress && showPublicKey {
+			return errors.New("cannot use both --address and --public-key at once")
+		}
+		if outputSet && (showAddress || showPublicKey) {
+			return errors.New("cannot use --output with --address or --public-key")
+		}
+		if showAddress {
+			printKeyAddress(info)
+			return nil
+		}
+		if showPublicKey {
+			printPubKey(info)
+			return nil
+		}
+
+		printInfo(info)
+		return nil
 	},
 }
 
+func init() {
+	showKeysCmd.Flags().Bool(client.FlagAddress, false, "output the address only (overrides --output)")
```

Comment:
Why not have a single `--text-format` flag, which is set to `NAME:\tTYPE:\tADDRESS:\t\t\t\t\t\tPUBKEY` by default? Then everybody can choose whatever text format they want.

Reply:
The objective of this output format is to allow piping to other commands, so we'd need to add additional options to not show that header.